admin/class-acc_user_importer-admin.php

```diff
@@ -510,8 +525,93 @@ private function proccess_user_data ( $users ) {
 		
 		return $api_response;
 	}
+
+
+	/**
+	 * Returns True if the user is expired.
+	 * If the user has no 'expiry' field, it must be a stale entry in the database.
+	 * So it is considered expired.
+	 */
+	private function is_user_expired ($user) {
+		if (empty($user->expiry)) {
+			$this->log_dual("user $user->ID $user->display_name has no expiry, consider expired");
+			return true;
+		}
+
+		if ($user->expiry < date("Y-m-d")) {
+			//$this->log_dual("user $user->ID $user->display_name expiry=$user->expiry is expired");
+			return true;
+		}
+		//$this->log_dual("user $user->ID $user->display_name expiry=$user->expiry is valid");
+		return false;
+	}
 	
 
+	/**
+	 * Go over our local user database and see who has an expired membership.
+	 */
+	private function proccess_expiry () {
+		$GLOBALS['acc_logstr'] = "";		//Clear the API response log string
+
+		// Get user-configurable option values
+		$options = get_option('accUM_data');
+
+		$this->log_dual("Now looking for expired members.");
+
+		//create response object
+		$api_response = Array();
+		$db_users = get_users("all_with_meta");
```

Comment:
I have to see if this works in our system, I am expecting it to fetch a lot of data and we might run into memory/timeout issues on our small server :sweat_smile: Should be OK for now though.

On a side-note, shouldn't this be `get_users(['fields' => 'all_with_meta'])`?

Reply:
Yes, I see. Maybe we can just get the list of users and then loop over each one and get all the specific user data.  For now I am just fixing as per your side-note comment, which I agree with.

Reply:
That was my thought too, but I think that would be premature optimization at this point. I'll know if it causes issues as soon I'll start testing in staging.